lane.py

```diff
@@ -550,6 +550,8 @@ def __init__(self,
         self.library_id = library.id
         self.collection_ids = [
             collection.id for collection in library.collections
+        ] + [
+            collection.parent_id for collection in library.collections if collection.parent_id
```

Comment:
I think it would be better to defined a method on Library that returned all its collections, recursively, and did `[c.id for collection in method()]` here. This can't be the only place we need all the collections. (Although the Lane stuff is centralized well enough that it might be.)